include/seqan3/alphabet/quality/phred42.hpp

```diff
@@ -0,0 +1,297 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \author Marie Hoffmann <marie.hoffmann AT fu-berlin.de>
+ * \brief Contains seqan3::phred42 quality scores.
+ */
+
+#pragma once
+
+#include <cassert>
+
+// ------------------------------------------------------------------
+// phred42
+// ------------------------------------------------------------------
+
+namespace seqan3
+{
+
+/*!\brief The sequence quality alphabet from '!' to 'J'.
+ * \ingroup quality
+ *
+ * \details
+ * The phred42 quality alphabet represents the zero-based phred score range
+ * [0..41] mapped to the ASCII range ['!' .. 'J']. It therefore can
+ * represent the Illumina 1.8+ standard and the original Sanger score. If you
+ * that larger phred scores are not mapped to 41, use the larger type phred63.
+ *
+ *~~~~~~~~~~~~~~~{.cpp}
+ *     phred42 phred;
+ *     phred = 2;
+ *     std::cout << phred.to_phred() << "\n";  // 2
+ *     std::cout << phred.to_char() << "\n";  // '#'
+ *     std::cout << phred.to_rank() << "\n";  // 2
+ *     phred = 49;
+ *     std::cout << phred.to_phred() << "\n";  // converted down to 41
+ *     // this
+ *     phred42 phred{dna4::A};
+ *     // doesn't work:
+ *     // phred42{4};
+ *~~~~~~~~~~~~~~~
+ */
+struct phred42
+{
+    /*!\name Member types
+    * \{
+    */
+    //!\brief The 0-based integer representation of a quality score.
+    using rank_type = uint8_t;
+
+    //!\brief The integer representation of a quality score assignable with =operator.
+    using phred_type = uint8_t;
+
+    //!\brief The '!'-based character representation of a quality score.
+    using char_type = char;
+    //!\}
+
+    //!\privatesection
+    /*!\name Member variables.
+    * \{
+    */
+    //!\brief The internal 0-based rank value.
+    rank_type _value;
+
+    //!\brief The projection offset between char and rank score representation.
+    static constexpr char_type offset_char{'!'};
+    //!\}
+
+    //!\brief Value assignment with implicit compatibility to inner type.
+    constexpr phred42 & operator=(phred_type const c)
+    {
+        assert(c < max_value_size);
+        _value = (c < value_size) ? c : value_size - 1;
+        return *this;
+    }
```

Comment:
This has to be removed, see above!

Reply:
You mean the complete `operator=` since assignment is done via the explicit assignment and conversion operators, right?

Reply:
Yes!